src/main/java/net/neoforged/neoforge/client/extensions/common/IClientItemExtensions.java

```diff
@@ -166,6 +170,39 @@ default boolean shouldSpreadAsEntity(ItemStack stack) {
         return true;
     }
 
+    /**
+     * Called when armor layers are rendered by {@link net.minecraft.client.renderer.entity.layers.HumanoidArmorLayer}.
+     * <p>
+     * Allows custom dye colors to be specified per-layer; default vanilla behavior allows for only a single dye color
+     * (specified by the {@link net.minecraft.core.component.DataComponents#DYED_COLOR} data component) for all layers.
+     * <p>
+     * Returning 0 here will cause rendering of this layer to be skipped entirely; this is recommended if the layer
+     * doesn't need to be rendered for a particular armor slot.
+     *
+     * @param stack         the armor item stack being rendered
+     * @param entity        the entity wearing the armor
+     * @param layer         the armor layer being rendered
+     * @param layerIdx      an index into the list of layers for the {@code ArmorMaterial} used by this item
+     * @param fallbackColor the return value of {@link #getDefaultDyeColor(ItemStack)}, passed as a parameter for
+     *                      performance
+     * @return a custom color for the layer, in ARGB format, or 0 to skip rendering
+     */
+    default int getLayerTintColor(ItemStack stack, LivingEntity entity, ArmorMaterial.Layer layer, int layerIdx, int fallbackColor) {
```

Comment:
If this only applies to armor layers, I'd suggest making the method name more specific to indicate that, possibly `getArmorLayerTintColor`?

Reply:
Yeah, works for me